### ISSUE TITLE
Enhancement: Switch profiles automatically based on the last input device

### ISSUE THREAD

**Issue by JukePlz** [KernelZechs/open-joystick-display#37]
### Problem: 
Certain games requiere several input devices to play, OJD currently requieres the user to switch manually to different profiles to show those inputs. This sugestion will cover different use cases and detail possible solutions and how they could be handled by UI.

### What this enhancement doesn't cover:
This feature request does **not** include simultaneous input devices, like first person shooters played with mouse and keyboard, or local multiplayer games with gamepads for different players. It only aims to improve the feature for streamers that need different input devices shown **in rotation**, and increase usability for OJD users that have many gamepads.

### Considerations:
This enhancement doesn't directly depend on #36 but benefits from it. Also, it doesn't depend on keyboard/mouse getting implemented as input devices and themes but it's considerably more useful with them, therefore I'm including examples use cases with them below.

**Use case 1 (without kb/mouse implementation):**
- User has connected a NES gamepad on port 1.
- User has connected a PS3 controller on port 2.
- User starts NES_GAME, picks up the NES gamepad and presses any button --> OJD automatically switches to the NES gamepad profile.
- User starts PS3_GAME, picks up the PS3 gamepad and presses any button --> OJD automatically switches to the PS3 gamepad profile.

**Use case 2 (with future kb/mouse implementation): Xanadu Next**
https://www.twitch.tv/videos/264059671?t=
In the linked video the speedrunner makes use of 3 input devices, a gamepad, mouse and keyboard.
- Shop navigation and equipment/inventory management requiere mouse input.
- General gameplay, movement, attack and menu navigation is done on gamepad input.
- Switching abilities/spells directly is done with keyboard keys F1, F2, F3, F4

_[While the game is playable with only gamepad, speedrunners or even regular users may still prefer the speed of directly selecting spell slots and the mouse precision/speed instead of analog stick cursor.]_

**Use case 3 (with future kb/mouse implementation): Elite Dangerous/flight sims**

Elite and flight sims use hundreads of functions for controls, populating gamepads by default in every posible combination, including using buttons+d-pad as modifiers for extra functions. This makes it imposible for experienced players to not use alternative input from other devices like a keyboard for other game commands that don't fit the gamepad layout.

Requiered UI elements to control the feature: Checkbox
![button](https://user-images.githubusercontent.com/8497884/73009708-d1873a80-3def-11ea-978f-4f647ac50b57.png)

**Note**: The state of the autoswitch button should be an invidual profile property, not a global setting. This allows users to exclude from autoswitching mouse/keyboard or any other devices they don't want shown to the stream.

**What happens when there is no input?**
The last input device is always the active profile. If the user pressed X button on a playstation gamepad that profile will remain active until another autoswitch enabled profile is trigged by other device.

### Possible issues and other considerations:
- Mouse autoswitching should only be done on mouse button input, not on cursor move. As the cursor is already visible without aid of OJD.
- I'm unaware of the feasibility of detecting different controller UIDs/hardware device ID in a multiplatform way. I don't know if Electron or RetroSpy offer this information.
- Multiple controllers of the same type may run into the issue of getting mixed up, this should be a rare case as user shouldn't really need different profiles for the same controller type.

Worst case scenario: profile switch only supports one gamepad, keyboard and mouse but not switching between different gamepads.

**Comment by JukePlz** [comment]
@KernelZechs I have made this a separate issue in the tracker, to not derail Terminounet sugestion. I've tried to make it as descriptive as possible, if you still want a flowchart I can provide that in a comment here.

**Comment by KernelZechs** [comment]
This is comprehensive and makes sense. So basically you want it to monitor which input is being used and switch to that profile automatically when actions are used for that input device without having to do it manually or have multiple instances opened.

This would be better suited for the v3 code base since I've completely re-written the way the state handlers work (I'm using react instead so easier to do this overall) and would make this easier to do so I would rather program this functionality there than in the 2.x codebase. As I continue my work on 3.x I'll keep this in mind as a feature.

**Comment by Darth-Vader0** [comment]
Good day, congratulations and thank you for this great program.

I comment that the program does not recognize the lever of the Xbox 360 Fightstick Madcatz "Tournament edition". The buttons had no problem.

Will there be any solution to this problem?

**Comment by KernelZechs** [comment]
@Darth-Vader0 This is the first I'm hearing about this. Please make a bug ticket and follow the instructions in the template and we can talk there.

**Comment by arturmiller79** [comment]
@KernelZechs would you be interest on paid custom project? I want program in PC to read XY(like yours do) of racing wheel, then convert this XY to XY meter to then convert to Latitude and longitude to then convert to NMEA 0183(like gps antena) message to output this in COM port at choice .  

we manufacture gps guidance for agriculture and i want to make gaming cockpit with farmsimulator19 and when the client drive in the game same movement happens in my console( i will recieve it by rs232) so I can use it for training.

**Comment by KernelZechs** [comment]
I am no longer working on this project. This project will be archived. You can find more info here: http://kernelzechs.com/open-joystick-display/
